cartography/intel/crxcavator.py

```diff
@@ -0,0 +1,210 @@
+    :param api_and_parameters: Query string for the API including the required parameters
+    :return: Returns JSON text blob for the API called. API spec is at https://crxcavator.io/apidocs
+    """
+    uri = CRXcavator_API_BASE_URL + api_and_parameters
+    data = requests.get(
+        uri,
+        headers={'Accept': 'application/json',
+                 'API-Key': CRXcavator_API_KEY})
+    if data.ok:
+        return data.json()
+    else:
+        logger.error('Error during CRXcavator API call {}'.format(data.status_code))
+        return None
+
+
+def ingest_extensions(session, update_tag):
+    """
+    Ingests the extension details into Neo4J
+    :param session: Neo4J session object for server communication
+    :param update_tag: Timestamp used to determine data freshness
+    :return: None
+    """
+    # the JSON returned from the CRXcavator API does not return well formatted objects
+    # instead, each object is named after it's key, making enumeration more difficult
+    # will build a cleaner object for import into graph
+
+    _data_index = 1
+    _row_index = 0
+
+    extension_json = get_extensions()
+    if not extension_json:
+        logger.error('No extensions data returned')
+        return
+    ingestion_cypher = """
+    UNWIND {ExtensionsData} as extension
+    MERGE (e:ChromeExtension{id: extension.id})
+    ON CREATE SET
+    e.extension_id = extension.extension_id,
+    e.version = extension.version,
+    e.firstseen = timestamp()
+    SET
+    e.extcalls = extension.extcalls,
+    e.risk_total = extension.risk_total,
+    e.risk_metadata = extension.risk_metadata,
+    e.address = extension.address,
+    e.email = extension.email,
+    e.icon = extension.icon,
+    e.last_updated = extension.last_updated,
+    e.name = extension.name,
+    e.offered_by = extension.offered_by,
+    e.permissions_warnings = extension.permissions_warnings,
+    e.privacy_policy = extension.privacy_policy,
+    e.rating = extension.rating,
+    e.rating_users = extension.rating_users,
+    e.short_description = extension.short_description,
+    e.size = extension.size,
+    e.support_site = extension.support_site,
+    e.users = extension.users,
+    e.website = extension.website,
+    e.type = extension.type,
+    e.price = extension.price,
+    e.offline_enabled = extension.offline_enabled,
+    e.lastupdated = {UpdateTag}
+    """
+
+    extensions = []
+    for extension in extension_json.items():
+        details = extension[_data_index][_row_index]
+        if not details:
+            continue
+        extension_id = details['extension_id']
+        version = details['version']
+        data = details['data']
+        # permissions warnings can be a list or null, if list concat to a single string
+        permission_warnings = data['webstore'].get('permission_warnings')
+        if permission_warnings:
+            permission_warnings = ', '.join(permission_warnings)
+        extensions.append({
```

Comment:
Please add some example return data to `cartography/tests/data/` so that we can some tests on ingesting crxcavator data.  Something similar to https://github.com/lyft/cartography/blob/master/tests/data/gcp/compute.py

Reply:
done

Reply:
Now that there's sample data, can you please add an integration test for your `load_*` functions? :)  Basically copy the format of [this](https://github.com/lyft/cartography/blob/master/tests/integration/cartography/intel/aws/test_dynamodb.py)

Reply:
integration test added